Tasks/VsTestPlatformToolInstaller/vstestplatformtoolinstaller.ts

```diff
@@ -6,230 +6,428 @@ import * as os from 'os';
-        tl.warning(tl.loc('NugetErrorCode', result.code));
+        debugLog('304');
+        warningLog(tl.loc('NugetErrorCode', result.code));
+        debugLog('306');
         throw new Error(tl.loc('ListPackagesFailed', result.code, result.stderr, result.stdout));
     }
+    debugLog('310');
 
     const listOfPackages = result.stdout.split('\r\n');
+    debugLog('313');
     let version: string;
+    debugLog('315');
 
     // Nuget returns latest vesions of all packages that match the given name, we need to filter out the exact package we need from this list
+    debugLog('318');
     listOfPackages.forEach(nugetPackage => {
+        debugLog('320');
         if (nugetPackage.split(' ')[0] === packageName) {
+            debugLog('322');
             version = nugetPackage.split(' ')[1];
+            debugLog('324');
             return;
         }
+        debugLog('328');
     });
+    debugLog('330');
 
     return version;
 }
 
 async function acquireAndCacheVsTestPlatformNuget(testPlatformVersion: string): Promise<string> {
+    debugLog('338');
     testPlatformVersion = toolLib.cleanVersion(testPlatformVersion);
+    debugLog('340');
     const nugetTool = tl.tool(path.join(__dirname, 'nuget.exe'));
+    debugLog('342');
     let downloadPath = tl.getVariable('Agent.TempDirectory');
+    debugLog('344');
 
     // Ensure Agent.TempDirectory is set
+    debugLog('347');
     if (!downloadPath) {
+        debugLog('349');
         throw new Error('Expected Agent.TempDirectory to be set');
     }
+    debugLog('353');
 
     // Call out a warning if the agent work folder path is longer than 50 characters as anything longer may cause the download to fail
+    debugLog('356');
     // Note: This upper limit was calculated for a particular test platform package version and is subject to change
+    debugLog('358');
     if (tl.getVariable('Agent.WorkFolder') && tl.getVariable('Agent.WorkFolder').length > 50) {
-        tl.warning(tl.loc('AgentWorkDirectoryPathTooLong'));
+        debugLog('360');
+        warningLog(tl.loc('AgentWorkDirectoryPathTooLong'));
+        debugLog('362');
     }
+    debugLog('364');
 
     // Use as short a path as possible due to nested folders in the package that may potentially exceed the 255 char windows path limit
+    debugLog('367');
     downloadPath = path.join(downloadPath, 'VsTest');
+    debugLog('369');
     nugetTool.line('install ' + packageName + ' -Version ' + testPlatformVersion + ' -Source ' + packageSource + ' -OutputDirectory "' + downloadPath + '" -NoCache -DirectDownload');
+    debugLog('371');
 
-    tl.debug(`Downloading Test Platform version ${testPlatformVersion} from ${packageSource} to ${downloadPath}.`);
+    debugLog(`Downloading Test Platform version ${testPlatformVersion} from ${packageSource} to ${downloadPath}.`);
+    debugLog('374');
     let startTime = perf();
+    debugLog('376');
     const resultCode = await nugetTool.exec();
```

Comment:
let resultCode:number ??
Should you fail on stdErr:true?  (see shellscript task).  Ask nuget if stderr means error (some tools write status to stderr like git)

Reply:
Will add the stderr option once i get this clarified about nuget.exe